packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -302,12 +302,12 @@ const ButtonBase = React.forwardRef(function ButtonBase(props, ref) {
       {...other}
     >
       {children}
-      {!disableRipple && !disabled ? (
-        <NoSsr>
-          {/* TouchRipple is only needed client-side, x2 boost on the server. */}
+      <NoSsr>
+        {!disableRipple && !disabled && (
+          /* TouchRipple is only needed client-side, x2 boost on the server. */
           <TouchRipple ref={rippleRef} center={centerRipple} {...TouchRippleProps} />
-        </NoSsr>
-      ) : null}
+        )}
```

Comment:
Make sure that `null` is not a valid child for `<NoSsr />`

Reply:
Maybe we could have NoSsr accept null children? It seems we have found a good use case.

Reply:
I don't see `null` here. It gets `false | ReactElement`.